### PR TITLE
Add examples on how to embed a Flutter screen into an existing SwiftUI app

### PR DESCRIPTION
**Pull request by leighajarett**
_Description of what this PR is changing or adding, and why:_

Developers are increasingly using SwiftUI, so it's important to show examples of how to embed Flutter into an existing SwiftUI app. I went through and tested the existing workflow and also added a few small things to project setup to make it easier, and to adapt to some new navigation within Xcode. 

_Issues fixed by this PR (if any):_ Fixes  https://github.com/flutter/website/issues/7039  and https://github.com/flutter/website/issues/6998 and https://github.com/flutter/flutter/issues/63051

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 1]
```diff
@@ -44,7 +44,43 @@ the app delegate.
...
+    // Used to connect plugins (only if you have plugins with iOS platform code).
+    GeneratedPluginRegistrant.register(with: self.npsFlutterEngine);
+  }
+}
+
+@main
+struct MyApp: App {
+  // flutterDependencies will be injected using EnvironmentObject.
+  @StateObject var flutterDependencies = FlutterDependencies()
+    var body: some Scene {
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:75]: indentation

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 2]
```diff
@@ -44,7 +44,43 @@ the app delegate.
...
+ **In `MyApp.swift`:**
+ <!--code-excerpt "MyApp.swift" title-->
+ ```swift
+import SwiftUI
+import Flutter
+// Used to connect plugins (only if you have plugins with iOS platform code).
+import FlutterPluginRegistrant
+
+class FlutterDependencies: ObservableObject {
+  let npsFlutterEngine = FlutterEngine(name: "my flutter engine")
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:62]: nit: maybe just `flutterEngine`? readers may not have the context of newsfeed app (iirc this nps is from the news feed demo app?)

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 3]
```diff
@@ -107,13 +143,56 @@ class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate.
 
 ### Show a FlutterViewController with your FlutterEngine
 
+{% samplecode vc %}
+
+{% sample SwiftUI %}
+The following example shows a generic `ContentView` with a
+`Button` hooked to present a [`FlutterViewController`][].
+The `FlutterViewController` uses the `FlutterEngine` instance
+passed in as an `EnvironmentObject`.
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:152]: Nit: This sounds like `FlutterViewController` requires an environment object. 
Also it's the FlutterDependencies, not flutter engine that gets injected. 

Maybe split into 2 sentences to make it more clear: 
1. the flutter view controller's constructor takes the pre-warmed engine as an argument 
2. this particular example uses environment object to inject flutter dependencies, which contains the flutter engine (this sentence is not as important, since you can also use singleton instead of dependency injection).

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 4]
```diff
@@ -107,13 +143,56 @@ class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate.
...
+<!--code-excerpt "ContentView.swift" title-->
+```swift
+import SwiftUI
+import Flutter
+
+struct ContentView: View {
+  // Flutter engine is passed in as an EnvironmentObject.
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:159]: similar here, technically it's the flutter dependencies, not flutter engine.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 5]
```diff
@@ -107,13 +143,56 @@ class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate.
...
+    guard
+      let windowScene = UIApplication.shared.connectedScenes
+        .first(where: { $0.activationState == .foregroundActive && $0 is UIWindowScene }) as? UIWindowScene,
+      let window = windowScene.windows.first(where: \.isKeyWindow),
+      let rootViewController = window.rootViewController
+    else { return }
+
+    // Create the FlutterViewController.
+    let flutterViewController = FlutterViewController(
+      engine: flutterDependencies.npsFlutterEngine,
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:180]: IIRC "nps" is from the news feed project, so readers of this doc may not have context.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 6]
```diff
@@ -201,7 +285,41 @@ To let the `FlutterViewController` present without an existing
...
 
 {% samplecode no-engine-vc %}
-{% sample Swift %}
+{% sample SwiftUI %}
+```swift
+import SwiftUI
+import Flutter
+
+struct ContentView: View {
+  // No EnvironmentObject needed.
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:294]: nit: remove? it may need environment object for other dependencies.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 7]
```diff
@@ -239,6 +357,87 @@ The `FlutterAppDelegate` performs functions such as:
...
+import Flutter
+import FlutterPluginRegistrant
+
+class AppDelegate: FlutterAppDelegate, ObservableObject {
+  let flutterEngine = FlutterEngine(name: "my flutter engine")
+
+  override func application(
+    _ application: UIApplication,
+    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+      // Runs the default Dart entrypoint with a default Flutter route.
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:377]: not sure if i understand this comment. seems like a in-depth topic that may not be suitable for an integration guide.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 8]
```diff
@@ -239,6 +357,87 @@ The `FlutterAppDelegate` performs functions such as:
...
+      flutterEngine.run();
+      // Used to connect plugins (only if you have plugins with iOS platform code).
+      GeneratedPluginRegistrant.register(with: self.flutterEngine);
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:380]: should we cover plugins? do we expect developers at this stage to know what plugins are?

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 9]
```diff
@@ -239,6 +357,87 @@ The `FlutterAppDelegate` performs functions such as:
...
+      return true;
+    }
+}
+
+@main
+struct MyApp: App {
+//  Use this property wrapper to tell SwiftUI
+//  it should use the AppDelegate class for the application delegate
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:388]: nit: maybe remove? we don't have to teach swiftui here

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 10]
```diff
@@ -239,6 +357,87 @@ The `FlutterAppDelegate` performs functions such as:
...
+    guard
+      let windowScene = UIApplication.shared.connectedScenes
+        .first(where: { $0.activationState == .foregroundActive && $0 is UIWindowScene }) as? UIWindowScene,
+      let window = windowScene.windows.first(where: \.isKeyWindow),
+      let rootViewController = window.rootViewController
+    else { return }
+
+    // Create the FlutterViewController.
+    let flutterViewController = FlutterViewController(
+      engine: flutterEngine,
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:428]: nit: can just be `engine: appDelegate.flutterEngine`

File: src/development/add-to-app/ios/project-setup.md [hunk 11]
```diff
@@ -478,3 +504,5 @@ You can now [add a Flutter screen][] to your existing application.
 [VS Code]: {{site.url}}/development/tools/vs-code
 [XCFrameworks]: {{site.apple-dev}}/documentation/xcode_release_notes/xcode_11_release_notes
 [Xcode installed]: {{site.url}}/get-started/install/macos#install-xcode
+[News Feed app]: https://github.com/flutter/put-flutter-to-work/tree/022208184ec2623af2d113d13d90e8e1ce722365
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/project-setup.md:507]: do we want po provide 2 examples? (one with app delegate and the other without)

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 12]
```diff
@@ -239,6 +357,87 @@ The `FlutterAppDelegate` performs functions such as:
   (which can only be detected in the AppDelegate) to
   Flutter for scroll-to-top behavior.
 
+### Creating a FlutterAppDelegate subclass
```
Comment (hellohuanlin) [inline src/development/add-to-app/ios/add-flutter-screen.md:360]: it's very possible that they already have an app delegate. maybe also document what to do if you can't inherit from flutter app delegate? iirc they will have to copy some functionalities from flutter app delegate over.

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 2 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:62]: Oops yeah my bad, thanks!

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 3 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:152]: Good call, thanks!

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 4 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:159]: Thanks!

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 10 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:428]: Sounds good

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 9 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:388]: I just thought it might be useful in case people are not using app delegate with SwiftUI, since it isn't required

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 8 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:380]: I can maybe link to the plugin docs

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 9 above)
Comment (hellohuanlin) [reply src/development/add-to-app/ios/add-flutter-screen.md:388]: gotcha

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 6 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:294]: Sounds good

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 7 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:377]: This comment was used in the existing Swift example so I was just trying to be consistent. The section near the bottom of this doc talks through the entry points so I think it's fine to leave

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 12 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:360]: That seems to be covered in the next section, but the example is Objective-C. I'm not really sure what the Swift equivalent looks like. Could you provide some guidance?

File: src/development/add-to-app/ios/project-setup.md (hunk 11 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/project-setup.md:507]: I imagine that someone could adapt the example based on the inline code we provide, but I'm sure it wouldn't hurt to have more examples. I don't think it makes sense to prioritize creating a full example right now, but maybe we can revisit after some of the other things (extensions, companion app, etc) have been added? WDYT?

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 7 above)
Comment (hellohuanlin) [reply src/development/add-to-app/ios/add-flutter-screen.md:377]: gotcha. my flutter knowledge is limited, but i guess it's fine if it's already there.

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 12 above)
Comment (hellohuanlin) [reply src/development/add-to-app/ios/add-flutter-screen.md:360]: > That seems to be covered in the next section, but the example is Objective-C.

can you point me there? i can help to translate it into swift.

File: src/development/add-to-app/ios/project-setup.md (hunk 11 above)
Comment (hellohuanlin) [reply src/development/add-to-app/ios/project-setup.md:507]: sounds good

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 12 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:361]: Yes, the current section is here: https://docs.flutter.dev/development/add-to-app/ios/add-flutter-screen?tab=engine-objective-c-tab#using-the-flutterappdelegate

In the new version of the doc it's under a subheader ### If you can't directly make FlutterAppDelegate as subclass

Comment (hellohuanlin) [reply src/development/add-to-app/ios/add-flutter-screen.md:367]: I translated this section to Swift here: 
https://github.com/hellohuanlin/put-flutter-to-work/commit/7515c37309cc839d24d969bf2f677bb897c49311

Also note that the existing objc section about the status bar seems outdated: https://github.com/flutter/flutter/issues/63051

So i did not translate that.

Comment (hellohuanlin) [reply src/development/add-to-app/ios/add-flutter-screen.md:367]: @cyanglaz may know what happens to this status bar API.

Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:367]: I don't think I have access to your version of put-flutter-to-work, can you grant me access?

Comment (hellohuanlin) [reply src/development/add-to-app/ios/add-flutter-screen.md:367]: Done

Comment (cyanglaz) [reply src/development/add-to-app/ios/add-flutter-screen.md:367]: It was removed in https://github.com/flutter/engine/pull/16820, not sure about the details, but it looks like people don't need to use this API anymore. @gaaclarke to confirm.

Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:367]: Cool thanks, I'm going to take it out for now.

**Comment by sfshaza2** [comment]
Ooh. I didn't mean to

File: src/development/add-to-app/ios/project-setup.md [hunk 13]
```diff
@@ -4,10 +4,31 @@ short-title: Integrate Flutter
...
+  and all Flutter plugins.** Here, you manually embed the frameworks,
+  and update your existing application's build settings in Xcode.
+  This can be useful for teams that don't want to require every developer
+  to have the Flutter SDK and Cocoapods installed locally.
+
+1. **Create frameworks for your compiled Dart code,
+  and all Flutter plugins. Create a Pod library for the Flutter engine.** 
+  With this option,  you embed the frameworks for your application
+  and the plugins in Xcode, but distribute the
+  Flutter engine as a Cocoapods podspec.
```
Comment (jmagman) [inline src/development/add-to-app/ios/project-setup.md:25]:
```suggestion
  Flutter engine as a CocoaPods podspec.
```

File: src/development/add-to-app/ios/project-setup.md [hunk 14]
```diff
@@ -4,10 +4,31 @@ short-title: Integrate Flutter
...
+  the source each time the app is built. (Recommended.)
+
+1. **Create frameworks for the Flutter engine, your compiled Dart code,
+  and all Flutter plugins.** Here, you manually embed the frameworks,
+  and update your existing application's build settings in Xcode.
+  This can be useful for teams that don't want to require every developer
+  to have the Flutter SDK and Cocoapods installed locally.
+
+1. **Create frameworks for your compiled Dart code,
+  and all Flutter plugins. Create a Pod library for the Flutter engine.**
```
Comment (jmagman) [inline src/development/add-to-app/ios/project-setup.md:22]: I don't think "Pod library" is quite right, maybe:
```suggestion
  and all Flutter plugins. Use CocoaPods for the Flutter engine.** 
```

File: src/development/add-to-app/ios/project-setup.md [hunk 15]
```diff
@@ -4,10 +4,31 @@ short-title: Integrate Flutter
...
+
+1. **Create frameworks for the Flutter engine, your compiled Dart code,
+  and all Flutter plugins.** Here, you manually embed the frameworks,
+  and update your existing application's build settings in Xcode.
+  This can be useful for teams that don't want to require every developer
+  to have the Flutter SDK and Cocoapods installed locally.
+
+1. **Create frameworks for your compiled Dart code,
+  and all Flutter plugins. Create a Pod library for the Flutter engine.** 
+  With this option,  you embed the frameworks for your application
```
Comment (jmagman) [inline src/development/add-to-app/ios/project-setup.md:23]:
```suggestion
  With this option, embed the frameworks for your application
```

File: src/development/add-to-app/ios/project-setup.md [hunk 16]
```diff
@@ -174,7 +197,7 @@ end
 ```
 
 {{site.alert.note}}
-  The `flutter_post_install` method (recently added to Flutter),
+  The `flutter_post_install` method (added in Flutter 3.1.0),
```
Comment (jmagman) [inline src/development/add-to-app/ios/project-setup.md:200]: Thanks for changing this.

File: src/development/add-to-app/ios/project-setup.md [hunk 17]
```diff
@@ -225,19 +248,15 @@ as a dependency manager in your existing applications.
 You must run `flutter build ios-framework`
 every time you make code changes in your Flutter module.
 
-If you're using the previous
-[Embed with CocoaPods and Flutter tools][] method,
-you can skip these instructions.
-
 The following example assumes that you want to generate the
-frameworks to `some/path/MyApp/Flutter/`.
+frameworks to `some/path/Flutter/`.
```
Comment (jmagman) [inline src/development/add-to-app/ios/project-setup.md:252]: I kind of liked the old `MyApp` directory so you know it should be relative or inside your host app in a known location, not just floating somewhere in another directory.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 18]
```diff
@@ -107,13 +149,58 @@ class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate.
...
+`FlutterEngine` as an argument. `FlutterEngine` is passed in 
+as an `EnvironmentObject` via `flutterDependencies`.
+
+<!--code-excerpt "ContentView.swift" title-->
+```swift
+import SwiftUI
+import Flutter
+
+struct ContentView: View {
+  // Flutter dependencies are passed in as an EnvironmentObject.
```
Comment (sfshaza2) [inline src/development/add-to-app/ios/add-flutter-screen.md:167]:
```suggestion
  // Flutter dependencies are passed in an EnvironmentObject.
```

File: src/development/add-to-app/ios/project-setup.md [hunk 19]
```diff
@@ -85,22 +112,16 @@ embedding the module into your existing application with CocoaPods.
...
+After you have developed your Flutter module,
+you can embed it using the methods described at the top of the page.
 
 {{site.alert.note}}
-  Your app does not run on a simulator in Release mode because Flutter does not
-  yet support outputting x86/x86_64 ahead-of-time (AOT) binaries for your Dart
-  code. You can run in Debug mode on a simulator or a real device,
-  and Release on a real device.
+  You can run in Debug mode on a simulator or a real device,
+  and Release on a real device. Learn more at
```
Comment (sfshaza2) [inline src/development/add-to-app/ios/project-setup.md:120]:
```suggestion
  and Release on a real device. Learn more about
```

File: src/development/add-to-app/ios/project-setup.md [hunk 20]
```diff
@@ -131,9 +153,10 @@ some/path/
 ```
 
 If your existing application (`MyApp`) doesn't
-already have a Podfile, follow the
-[CocoaPods getting started guide][]
-to add a `Podfile` to your project.
+already have a Podfile, run `pod init` in the  
+`MyApp` directory to create one. 
+You can find more details on using 
+Cocoapods in the [CocoaPods getting started guide][].
```
Comment (sfshaza2) [inline src/development/add-to-app/ios/project-setup.md:159]:
```suggestion
CocoaPods in the [CocoaPods getting started guide][].
```

**Review by sfshaza2 — COMMENTED** [review]
@leighajarett, I had very few minor nits, but I do agree with @jmagman's feedback - at least where I added a thumbs up.  :D

**Comment by sfshaza2** [comment]
Once everything is resolved, it lgtm

File: src/development/add-to-app/ios/project-setup.md (hunk 16 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/project-setup.md:200]: No prob!

File: src/development/add-to-app/ios/project-setup.md (hunk 17 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/project-setup.md:252]: The reason why I moved it out was because I thought it wouldn't be directly inside your app codebase because I figured if other developers didn't have dart / flutter on their computer then it wouldn't make sense for them to bring the code in when they're working on the app. But I can definitely add it back!

Comment (jmagman) [reply src/development/add-to-app/ios/project-setup.md:252]: This is the path to the built Flutter/App/plugin frameworks, not the Flutter module containing the dart code.  Without these frameworks the app won't build, so they need to make it available to other developers that don't have Flutter installed somehow.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 21]
```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
...
+to your host app. 
 
 {% samplecode engine %}
 
-{% sample Swift %}
+{% sample SwiftUI %}
+As an example, we demonstrate creating a
+`FlutterEngine` inside an `ObservableObject`, 
+and passing that into a `View` using the 
+ `environmentObject()` modifier.
```
Comment (atsansone) [inline src/development/add-to-app/ios/add-flutter-screen.md:49]: **issue:** This section should indicate what's done where. This throws in a lot of information with little context.
```suggestion
In this example, we create a `FlutterEngine` object inside a SwiftUI `ObservableObject`. 
We then pass this `FlutterEngine` into a `ContentView` using the 
 `environmentObject()` property. 
```

**suggestion:** It would not hurt to link out to the [SwiftUI docs](https://developer.apple.com/documentation/swiftui/environmentobject) in this case to make sure the dev can follow along.

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 22]
```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
...
+and passing that into a `View` using the 
+ `environmentObject()` modifier. 
+
+ **In `MyApp.swift`:**
+ <!--code-excerpt "MyApp.swift" title-->
+ ```swift
+import SwiftUI
+import Flutter
+// Used to connect plugins (only if you have plugins with iOS platform code).
+import FlutterPluginRegistrant
```
Comment (atsansone) [inline src/development/add-to-app/ios/add-flutter-screen.md:57]: **question:** Which line does this modify? The previous or the following? It reads like it's the previous. If it is the `FlutterPluginRegistrant`, then move this comment to after that line.

**suggestion:** Consider the context. _What_ is used to connect plugins? Also, why use the parenthetical? It can be combined.

```suggestion
import Flutter
import FlutterPluginRegistrant
// The previous library connects plugins with iOS platform code to this app.
```

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 23]
```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
...
+
+class FlutterDependencies: ObservableObject {
+  let flutterEngine = FlutterEngine(name: "my flutter engine")
```
Comment (atsansone) [inline src/development/add-to-app/ios/add-flutter-screen.md:60]: **suggestion:** Don't label the variable with the same exact label as the Object. That makes it hard to follow. 

```suggestion
  let myFlutterEngine = FlutterEngine(name: "my flutter engine")
```

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 24]
```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
...
+  init(){
+    // Prepare a Flutter engine in advance.
```
Comment (atsansone) [inline src/development/add-to-app/ios/add-flutter-screen.md:62]: **question:** In advance of what?

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 25]
```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
...
+    flutterEngine.run()
+    // Used to connect plugins (only if you have plugins with iOS platform code).
```
Comment (atsansone) [inline src/development/add-to-app/ios/add-flutter-screen.md:64]: **repeat:** See [line 57](https://github.com/flutter/website/pull/7828/files#diff-16e6938e4d60842765926435d41283877a42dc43768665948e328d2b563739d2R57).

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 26]
```diff
@@ -67,7 +105,11 @@ class AppDelegate: FlutterAppDelegate { // More on the FlutterAppDelegate.
   }
 }
 ```
-{% sample Objective-C %}
+{% sample UIKit-ObjC %}
+As an example, we demonstrate creating a
+`FlutterEngine`, exposed as a property, on app startup in
+the app delegate.
```
Comment (atsansone) [inline src/development/add-to-app/ios/add-flutter-screen.md:111]: **repeat:** See [line 49](https://github.com/flutter/website/pull/7828/files#diff-16e6938e4d60842765926435d41283877a42dc43768665948e328d2b563739d2R49).

**Review by atsansone — CHANGES_REQUESTED** [review]
@leighajarett : I noted some consistency and clarity issues here along with a few questions. In general, I would check:

- Use labels for variables that are not the same as the Object type.
- Clarify what code the comments are clarifying.
- Simplify some of the language.

I stopped my comments with the first instance of an issue. The comment would apply to any copy that has a similar issue (like variable names).

File: src/development/add-to-app/ios/add-flutter-screen.md [hunk 27]
```diff
@@ -38,13 +38,51 @@ trade-offs of pre-warming an engine.
 ### Create a FlutterEngine
 
 The proper place to create a `FlutterEngine` is specific
-to your host app. As an example, we demonstrate creating a
-`FlutterEngine`, exposed as a property, on app startup in
-the app delegate.
+to your host app.
```
[the remark this answers is not in the document]
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:41]: I think we want to say what we're doing though - I think its important that we mention we're exposing FlutterEngine as an app delegate when the app starts up. Is there a better way to word it that includes this information?

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 22 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:57]: Hmm I'm more familiar with comments that go before the line that it is describing, but I can make it explicit that we are referring to FlutterPluginRegistrant

File: src/development/add-to-app/ios/project-setup.md (hunk 17 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/project-setup.md:252]: Ahh right, okay makes sense -- I'll change back!

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:41]: Actually never mind I see you add this below!

[the remark this answers is not in the document]
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:41]: okay!

File: src/development/add-to-app/ios/add-flutter-screen.md (hunk 23 above)
Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:60]: We use flutterEngine as the variable name in several examples that were already in this document and are available as demo apps in multiple places on GitHub, so I think it makes sense to leave it as is. Are you okay with that or would you still like me to change it?

Comment (atsansone) [reply src/development/add-to-app/ios/add-flutter-screen.md:58]: **reply:** Leave it. If you could create a new issue to address it later, that would be appreciated!

Comment (leighajarett) [reply src/development/add-to-app/ios/add-flutter-screen.md:58]: Done!

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Review by atsansone — APPROVED** [review]
